File: dec2vec.m
function phi=dec2vec(dec,n)
% Author: Charles Fox, Robotics Research Group, Oxford University
% Accessed: April 2022
%phi = dec2vec(dec,n)
%convert single dec state to vector representation
%n is max bits

cf_assert(dec<=2^n-1, 'dec too big for register size n!');

phi=zeros(2^n,1);
phi(dec+1)=1; %compensate for matlab indexing with +1

File: cpp_Implementation/dec2vec.cpp
#include <vector>

// Function to convert a single decimal state to vector representation.
std::vector<int> dec2vec(int dec, int n) {
    assert(dec <= pow(2, n)-1 && "dec too big for register size n!");
    std::vector<int> phi(pow(2, n), 0);
    phi[dec] = 1; 
    return phi;
}
